ENIGMAsystem/SHELL/Graphics_Systems/Direct3D11/DX11colors.cpp

```diff
@@ -85,6 +88,7 @@ void draw_set_color_rgba(unsigned char red,unsigned char green,unsigned char blu
 
 void draw_set_color_write_enable(bool red, bool green, bool blue, bool alpha)
 {
-
+	draw_batch_flush(batch_flush_deferred);
```

Comment:
Doesn't this function control whether we're writing colors into a VBO? Why does this require a flush?

Reply:
This controls the color buffer (depth buffer's best friend) of the framebuffer (nothing to do with the vertex data). If you turn off writing to e.g, the red channel of the color buffer, then you need to flush everything that was already drawn just before making the state change.

Please review the documentation for glColorMask:
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glColorMask.xhtml